Split integration tests into more parts

donoghuc opened the pull request:
Previously we were splitting in to 3 parts. The integration tests run in about 25 minutes. Without splitting the unit tests we see run times at about 20 minutes. Splitting integration tests across 6 workers instead of 3 puts the integration tests in that range. 

This is a very easy win as @yaauie had already built out the splitting functionality, this just tunes those knobs such that integration tests are no longer the bottleneck and instead run about as fast as the next slowest test set. 

The target here is seeing the entire feedback cycle for PR tests closer to 20 minutes rather than closer to 30 minutes with very little cognitive investment. 

Relates to https://github.com/elastic/ingest-dev/issues/5579

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/3358)
* Commit: 9747b4562b843436e22cd2cd4a58af7491303700

<!--buildkite-pr-comment-logstash-pull-request-pipeline
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/3358","number":3358,"commit":"9747b4562b843436e22cd2cd4a58af7491303700"}],"number":3358}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/3359)
* Commit: ec9336858cc8bd61471ee66bf5036b11908398b4

### History
* :green_heart: [Build #3358](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/3358) succeeded 9747b4562b843436e22cd2cd4a58af7491303700

<!--buildkite-pr-comment-logstash-pull-request-pipeline
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/3359","number":3359,"commit":"ec9336858cc8bd61471ee66bf5036b11908398b4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/3358","number":3358,"commit":"9747b4562b843436e22cd2cd4a58af7491303700"}],"number":3359}
buildkite-pr-comment-->

donoghuc commented:
Split 3 ways:
<img width="1728" height="1036" alt="image" src="https://github.com/user-attachments/assets/c40414da-1f7f-4c1e-b95d-73946e8a804e" />

Split 4 ways:
<img width="1728" height="1036" alt="image" src="https://github.com/user-attachments/assets/6a1bf132-3d7a-4d0e-ac99-7afad1461da8" />

Split 6 ways:
<img width="1728" height="1036" alt="image" src="https://github.com/user-attachments/assets/24e97c59-ae48-4685-8d9b-a00019cd330d" />

donoghuc commented:
Thanks @mashhurs I thought about that applying that pattern here as there is quite a bit of duplication. I'm for sure open to it in the future.
